src/components/Tool/FixPhoneNumbers/FixPhoneNumbers.tsx

```diff
@@ -189,258 +176,303 @@ const FixPhoneNumbers: React.FC<Props> = ({
-    const temp = { ...dataState };
-    dataState[personId].phoneNumbers[numberIndex].number = event.target.value;
-    setDataState(temp);
-  };
-
-  const handleDelete = (): void => {
-    const temp = { ...dataState };
-    const deleting = temp[deleteModalState.personId].phoneNumbers.splice(
-      deleteModalState.numberIndex,
-      1,
-    )[0];
-    deleting.destroy = true;
-    deleting.primary &&
-      (temp[deleteModalState.personId].phoneNumbers[0].primary = true); // If the deleted number was primary, set the new first index to primary
-    deleting.id &&
-      temp[deleteModalState.personId].toDelete.push({
-        destroy: true,
-        id: deleting.id,
-      }); //Only destroy the number if it already exists (has an ID)
-    setDataState(temp);
-    handleDeleteModalClose();
-  };
-
-  const handleAdd = (personId: string, number: string): void => {
-    const temp = { ...dataState };
-    temp[personId].phoneNumbers.push({
-      updatedAt: new Date().toISOString(),
-      number: number,
-      primary: false,
-      source: 'MPDX',
+    setDeleteModalState({
+      open: true,
+      personIndex,
+      numberIndex,
+      phoneNumber,
     });
-    setDataState(temp);
   };
 
-  const handleChangePrimary = (personId: string, numberIndex: number): void => {
-    const temp = { ...dataState };
-    temp[personId].phoneNumbers = temp[personId].phoneNumbers.map(
-      (number, index) => ({
-        ...number,
-        primary: index === numberIndex ? true : false,
-      }),
+  const handleBulkConfirm = async (values: FormValues) => {
+    const dataToSend = determineBulkDataToSend(
+      values?.people,
+      defaultSource ?? '',
     );
-    setDataState(temp);
-  };
 
-  const handleSourceChange = (event: SelectChangeEvent<string>): void => {
-    setDefaultSource(event.target.value);
-  };
+    if (!dataToSend.length) {
+      return;
+    }
 
-  const updatePhoneNumber = async (
-    personId: string,
-    name: string,
-    numbers: PhoneNumberData[],
-  ): Promise<void> => {
-    const attributes = [
-      {
-        phoneNumbers: numbers.map((phoneNumber) => ({
-          id: phoneNumber.id,
-          primary: phoneNumber.primary,
-          number: phoneNumber.number,
-          validValues: true,
-        })),
-        id: personId,
-      },
-    ];
     await updateInvalidPhoneNumbers({
       variables: {
         input: {
           accountListId,
-          attributes,
+          attributes: dataToSend,
         },
       },
       update: (cache) => {
-        cache.evict({ id: `Person:${personId}` });
+        data?.people.nodes.forEach((person: PersonInvalidNumberFragment) => {
+          cache.evict({ id: `Person:${person.id}` });
+        });
       },
-      onError() {
-        enqueueSnackbar(t(`Error updating ${name}'s phone numbers`), {
+      onError: () => {
+        enqueueSnackbar(t(`Error updating phone numbers`), {
           variant: 'error',
           autoHideDuration: 7000,
```

Comment:
Is there a good reason to customize the default auto-hide duration?

Reply:
I saw that too after I commented. It doesn't really matter, but it seems good to have consistent auto-hide durations, which is easier to achieve by using the default values.